### DISCUSSION TITLE
Unwanted behavior of checkboxes for selection in tables

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#2375] [PrimeFaces] [answered]
If I have already marked several lines in a list and then do not click on the exact checkbox to complete the selection, all previously selected lines are deselected and only the one line where I clicked next to it is marked. Is it possible that when you click next to it, not all other lines are deselected, but the selected line is added to the previously selected ones?

**Comment by melloware** [comment]
try `selectionRowMode="none"` that disables the row selection and allows it to only be on checkboxes.

https://primefaces.github.io/primefaces/vdldoc/p/dataTable.html#selectionRowMode

In PF14 its `selectionRowMode` in PF13- it was `rowSelectionMode` and had different values check the docs for 13.

**Reply by hgfrey** [reply]
I'm discussing this with my customer, but I think he won't want to do without selection across the rows.

**Reply by melloware** [reply]
OK then what is happening is if his mouse is outside the checkbox then it becomes a ROW click and thus the behavior you are seeing.

**Reply by hgfrey** [reply, marked as answer]
Yes.

I built a small JavaScript snippet that prevents clicking on the Selection TD without preventing the click on the Selectbox. This means that selection can now be made both via the lines and via the select boxes. I presented this to my customer and am curious to see what he will say about it.

```
$(".ui-selection-column").click(function (e) {
    if (e.target.tagName === 'TD') {
        return false;
    } else {
        return true;
    }
});
```

**Reply by melloware** [reply]
you can simplify to just...

```js
$(".ui-selection-column").click(function (e) {
   return e.target.tagName !== 'TD'
});

**Reply by hgfrey** [reply]
You're right. 

That was just a quick check to see if it worked. If my customer wants this solution, then I also have to check whether the selector is strong enough and whether it is sufficient to check for TD.
